aksel.nav.no/website/scripts/changelog.ts

```diff
@@ -40,21 +41,66 @@ export async function main() {
 
   const blockContentType = defaultSchema
     .get("blogPost")
-    .fields.find((field) => field.name === "body").type;
-
-  const blocks = htmlToBlocks(html, blockContentType, {
-    parseHtml: (htmlStr) => new JSDOM(htmlStr).window._document,
-  });
-  blocks.shift();
+    .fields.find((field: any) => field.name === "body").type;
 
-  client
-    .patch("75351662-ecb0-455a-b9a5-59f319e423fa")
-    .set({ content: blocks })
-    .commit()
-    .then(() => {
-      console.info("Hurray, updated changelog!");
-    })
-    .catch((err) => {
-      console.error("Oh no, the changelog-update failed: ", err.message);
+  // We go through the last 3 entries in case there has been changes,
+  // or in case previous attempts have failed.
+  for (let i = 0; i < 3; i++) {
+    const currentChangelogEntry = changelog[i];
+    const version = currentChangelogEntry.substring(
+      0,
+      currentChangelogEntry.indexOf("\n"),
+    );
+    const text = currentChangelogEntry
+      .substring(currentChangelogEntry.indexOf("\n") + 1)
+      .replaceAll("### ", ""); // Make package names regular text instead of headings
+    const heading = `Versjon ${version}`;
+    const blocks = htmlToBlocks(converter.makeHtml(text), blockContentType, {
+      parseHtml: (htmlStr) => new JSDOM(htmlStr).window._document,
     });
+
+    // Check for existing document
+    const existingDoc = await client.fetch(
+      `*[_type == "ds_endringslogg_artikkel" && endringstype == 'kode' && heading == $heading]`,
+      { heading },
```

Comment:
Isnt `slug` the main identifier, since only the slug needs to be unique in theory? 🤔